tests/integration/dataframe/test_dataframe.py

```diff
@@ -276,3 +277,60 @@ def test__modify_tables__returns_partial_success(self, client: DataFrameClient):
         assert response.modified_table_ids == [id]
         assert response.failed_modifications == [updates[1]]
         assert len(response.error.inner_errors) == 1
+
+    def test__read_and_write_data__works(self, client: DataFrameClient, create_table):
+        id = create_table(
+            models.CreateTableRequest(
+                columns=[
+                    models.Column(
+                        name="index",
+                        data_type=models.DataType.Int32,
+                        column_type=models.ColumnType.Index,
+                    ),
+                    models.Column(
+                        name="value",
+                        data_type=models.DataType.Float64,
+                        column_type=models.ColumnType.Nullable,
+                    ),
+                    models.Column(
+                        name="ignore_me",
+                        data_type=models.DataType.Bool,
+                        column_type=models.ColumnType.Nullable,
+                    ),
+                ]
+            )
+        )
+
+        frame = models.DataFrame(
+            columns=["index", "value", "ignore_me"],
+            data=[["1", "3.3", "True"], ["2", None, "False"], ["3", "1.1", "True"]],
+        )
+
+        client.append_table_data(
+            id, models.AppendTableDataRequest(frame=frame, end_of_data=True)
```

Comment:
Should we add an error case, either passing invalid data or appending to a table after marking it end of data?

Reply:
Added a test that tries to write to a column that doesn't exist